pallets/subtensor/src/swap.rs

```diff
@@ -92,6 +92,102 @@ impl<T: Config> Pallet<T> {
+    /// - There's not enough balance to pay for the swap.
+    ///
+    /// # Events
+    ///
+    /// Emits a `ColdkeySwapped` event when successful.
+    ///
+    /// # Weight
+    ///
+    /// Weight is tracked and updated throughout the function execution.
+    pub fn do_swap_coldkey(
+        origin: T::RuntimeOrigin,
+        old_coldkey: &T::AccountId,
+        new_coldkey: &T::AccountId,
+    ) -> DispatchResultWithPostInfo {
+        let caller = ensure_signed(origin)?;
+
+        // Ensure the caller is the old coldkey
+        ensure!(caller == *old_coldkey, Error::<T>::NonAssociatedColdKey);
+
+        let mut weight = T::DbWeight::get().reads(2);
+
+        ensure!(
+            old_coldkey != new_coldkey,
+            Error::<T>::NewColdKeyIsSameWithOld
+        );
+
+        // // Check if the new coldkey is already associated with any hotkeys
+        // ensure!(
+        //     !Self::coldkey_has_associated_hotkeys(new_coldkey),
+        //     Error::<T>::ColdKeyAlreadyAssociated
+        // );
+
+        let block: u64 = Self::get_current_block_as_u64();
+        // ensure!(
+        //     !Self::exceeds_tx_rate_limit(Self::get_last_tx_block(old_coldkey), block),
+        //     Error::<T>::ColdKeySwapTxRateLimitExceeded
+        // );
+
+        // Note: we probably want to make this free
+        let swap_cost = Self::get_coldkey_swap_cost();
+        ensure!(
+            Self::can_remove_balance_from_coldkey_account(old_coldkey, swap_cost),
+            Error::<T>::NotEnoughBalanceToPaySwapColdKey
+        );
+        let actual_burn_amount = Self::remove_balance_from_coldkey_account(old_coldkey, swap_cost)?;
+        Self::burn_tokens(actual_burn_amount);
+
+        // Swap coldkey references in storage maps
+        Self::swap_total_coldkey_stake(old_coldkey, new_coldkey, &mut weight);
+        Self::swap_stake_for_coldkey(old_coldkey, new_coldkey, &mut weight);
+        Self::swap_owner_for_coldkey(old_coldkey, new_coldkey, &mut weight);
+        Self::swap_total_hotkey_coldkey_stakes_this_interval_for_coldkey(
+            old_coldkey,
+            new_coldkey,
+            &mut weight,
+        );
+        Self::swap_subnet_owner_for_coldkey(old_coldkey, new_coldkey, &mut weight);
+        Self::swap_owned_for_coldkey(old_coldkey, new_coldkey, &mut weight);
+
+        // Transfer any remaining balance from old_coldkey to new_coldkey
+        let remaining_balance = Self::get_coldkey_balance(old_coldkey);
+        if remaining_balance > 0 {
+            Self::kill_coldkey_account(old_coldkey, remaining_balance)?;
+            Self::add_balance_to_coldkey_account(new_coldkey, remaining_balance);
+        }
+
+        Self::set_last_tx_block(new_coldkey, block);
+        weight.saturating_accrue(T::DbWeight::get().writes(1));
+
+        Self::deposit_event(Event::ColdkeySwapped {
+            old_coldkey: old_coldkey.clone(),
+            new_coldkey: new_coldkey.clone(),
```

Comment:
Don't exactly need to clone here, since the variables here aren't being used later on.

Reply:
I get this error when i remove the clone 

```rust
mismatched types
expected associated type `_`
         found reference `&_`rustcClick for full compiler diagnostic
swap.rs(184, 37): consider using clone here: `.clone()`
```